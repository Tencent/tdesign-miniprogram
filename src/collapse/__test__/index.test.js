import path from 'path';
import simulate from 'miniprogram-simulate';

describe('collapse', () => {
  const id = load(path.resolve(__dirname, `./index`));

  it(`: style && customStyle`, async () => {
    const comp = simulate.render(id);
    comp.attach(document.createElement('parent-wrapper'));

    const collapse = comp.querySelector('#base >>> .t-collapse');
    const collapsePanel = comp.querySelector('#first >>> .t-collapse-panel');

    // expect(comp.toJSON()).toMatchSnapshot();
    if (VIRTUAL_HOST) {
      expect(collapse.dom.getAttribute('style').includes(`${comp.data.style}; ${comp.data.customStyle}`)).toBeTruthy();
      expect(
        collapsePanel.dom.getAttribute('style').includes(`${comp.data.style}; ${comp.data.customStyle}`),
      ).toBeTruthy();
    } else {
      expect(collapse.dom.getAttribute('style').includes(`${comp.data.customStyle}`)).toBeTruthy();
      expect(collapsePanel.dom.getAttribute('style').includes(`${comp.data.customStyle}`)).toBeTruthy();
    }
  });

  it(':event', async () => {
    const comp = simulate.render(id);
    comp.attach(document.createElement('parent-wrapper'));

    const $base = comp.querySelector('#base');
    const $second = comp.querySelector('#second >>> .t-collapse-panel__title');
    const $transitionWrapper = comp.querySelector('#second >>> .t-collapse-panel__wrapper');

    $second.dispatchEvent('tap');
    $transitionWrapper.dispatchEvent('transitionend');

    await simulate.sleep();

    expect($base.instance.data.value.includes(1)).toBeTruthy();

    $second.dispatchEvent('tap');

    await simulate.sleep();

    expect($base.instance.data.value.includes(1)).toBeFalsy();

    // expandMutex = true
    comp.setData({ expandMutex: true });

    expect($base.instance.data.value.includes(0)).toBeTruthy();

    $second.dispatchEvent('tap');

    await simulate.sleep();

    expect($base.instance.data.value.includes(0)).toBeFalsy();
    expect($base.instance.data.value.includes(1)).toBeTruthy();
  });

  it(':defaultExpandAll', async () => {
    const comp = simulate.render(id);
    comp.attach(document.createElement('parent-wrapper'));

    const $expandAll = comp.querySelector('#expandAll');

    expect($expandAll.toJSON()).toMatchSnapshot();
  });

  it(':disabled', async () => {
    const comp = simulate.render(id);
    comp.attach(document.createElement('parent-wrapper'));

    comp.setData({ secondDisabled: true });

    await simulate.sleep();

    // const $base = comp.querySelector('#base');

    // expect($base.toJSON()).toMatchSnapshot();
  });
});
