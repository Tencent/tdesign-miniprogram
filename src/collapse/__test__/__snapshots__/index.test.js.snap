// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collapse :defaultExpandAll 1`] = `
<t-collapse
  id="expandAll"
>
  <wx-view
    class="class t-class t-collapse t-collapse--default"
    style=""
  >
    <t-collapse-panel>
      <wx-view
        class="class t-collapse-panel t-collapse-panel--bottom t-class"
        style=""
      >
        <wx-view
          ariaDisabled="{{false}}"
          ariaExpanded="{{true}}"
          ariaRole="button"
          class="t-collapse-panel__title"
          bind:tap="onClick"
        >
          <t-cell
            tClass="t-collapse-panel__header t-collapse-panel__header--bottom t-collapse-panel__header--expanded t-class-header"
            tClassHover="class-header-hover"
            tClassNote="class-note "
            tClassRightIcon="class-right-icon t-collapse-panel__arrow--bottom "
            tClassTitle="class-title "
          >
            <wx-view
              ariaLabel=""
              ariaRole=""
              class="class t-class t-cell  t-cell--middle"
              hoverClass=""
              hoverStayTime="70"
              style=""
              bind:tap="onClick"
            >
              <wx-view
                class="t-cell__left t-class-left"
              />
              <wx-view
                class="t-cell__title t-class-center"
              >
                <wx-view
                  class="t-cell__title-text t-class-title"
                >
                   first 
                </wx-view>
                <wx-view
                  class="t-cell__description t-class-description"
                />
              </wx-view>
              <wx-view
                class="t-cell__note t-class-note"
              />
              <wx-view
                class="t-cell__right t-class-right"
              >
                <t-icon
                  class=""
                  tClass="t-cell__right-icon t-class-right-icon"
                  bind:click=""
                >
                  <wx-view
                    ariaHidden="{{false}}"
                    ariaLabel=""
                    ariaRole=""
                    class="t-icon class t-class"
                    style=""
                    bind:tap="onTap"
                  >
                    <wx-label
                      class="t-icon-chevron-up t-icon-base"
                    />
                  </wx-view>
                </t-icon>
              </wx-view>
            </wx-view>
          </t-cell>
        </wx-view>
        <wx-view
          animation=""
          ariaHidden=""
          class="t-collapse-panel__wrapper"
        >
          <wx-view
            class="t-collapse-panel__content t-class-content"
          >
            
      
      
            内容1
          </wx-view>
        </wx-view>
      </wx-view>
    </t-collapse-panel>
    <t-collapse-panel>
      <wx-view
        class="class t-collapse-panel t-collapse-panel--bottom t-class"
        style=""
      >
        <wx-view
          ariaDisabled="{{false}}"
          ariaExpanded="{{true}}"
          ariaRole="button"
          class="t-collapse-panel__title"
          bind:tap="onClick"
        >
          <t-cell
            tClass="t-collapse-panel__header t-collapse-panel__header--bottom t-collapse-panel__header--expanded t-class-header"
            tClassHover="class-header-hover"
            tClassNote="class-note "
            tClassRightIcon="class-right-icon t-collapse-panel__arrow--bottom "
            tClassTitle="class-title "
          >
            <wx-view
              ariaLabel=""
              ariaRole=""
              class="class t-class t-cell  t-cell--middle"
              hoverClass=""
              hoverStayTime="70"
              style=""
              bind:tap="onClick"
            >
              <wx-view
                class="t-cell__left t-class-left"
              />
              <wx-view
                class="t-cell__title t-class-center"
              >
                <wx-view
                  class="t-cell__title-text t-class-title"
                >
                   second 
                </wx-view>
                <wx-view
                  class="t-cell__description t-class-description"
                />
              </wx-view>
              <wx-view
                class="t-cell__note t-class-note"
              />
              <wx-view
                class="t-cell__right t-class-right"
              >
                <t-icon
                  class=""
                  tClass="t-cell__right-icon t-class-right-icon"
                  bind:click=""
                >
                  <wx-view
                    ariaHidden="{{false}}"
                    ariaLabel=""
                    ariaRole=""
                    class="t-icon class t-class"
                    style=""
                    bind:tap="onTap"
                  >
                    <wx-label
                      class="t-icon-chevron-up t-icon-base"
                    />
                  </wx-view>
                </t-icon>
              </wx-view>
            </wx-view>
          </t-cell>
        </wx-view>
        <wx-view
          animation=""
          ariaHidden=""
          class="t-collapse-panel__wrapper"
        >
          <wx-view
            class="t-collapse-panel__content t-class-content"
          >
            
      
      
            内容2
          </wx-view>
        </wx-view>
      </wx-view>
    </t-collapse-panel>
  </wx-view>
</t-collapse>
`;
